config/main.py

```diff
@@ -231,6 +231,76 @@ def interface_name_is_valid(interface_name):
                     return True
     return False
 
+
+#
+# Use this method to validate unicast IPv4 address
+#
+def is_ip4_addr_valid(addr, display):
+    v4_invalid_list = [ipaddress.IPv4Address(unicode('0.0.0.0')), ipaddress.IPv4Address(unicode('255.255.255.255'))]
+    try:
+        ip = ipaddress.ip_address(unicode(addr))
+        if (ip.version == 4):
+            if (ip.is_reserved):
+                if display:
+                    click.echo ("{} Not Valid, Reason: IPv4 reserved address range.".format(addr))
+                return False
+            elif (ip.is_multicast):
+                if display:
+                    click.echo ("{} Not Valid, Reason: IPv4 Multicast address range.".format(addr))
+                return False
+            elif (ip in v4_invalid_list):
+                if display:
+                    click.echo ("{} Not Valid.".format(addr))
+                return False
+            else:
+                return True
+
+        else:
+            if display:
+                click.echo ("{} Not Valid, Reason: Not an IPv4 address".format(addr))
+            return False
+
+    except ValueError:
+        return False
+
+def vlan_id_is_valid(vid):
+    """Check if the vlan id is in acceptable range (between 1 and 4094)
+    """
+
+    if vid<1 or vid>4094:
+        return False
+
+    return True
+
+def vni_id_is_valid(vni):
+    """Check if the vni id is in acceptable range (between 1 and 2^24)
+    """
+
+    if (vni < 1) or (vni > 16777215):
+        return False
+
+    return True
+
+def is_vni_vrf_mapped(ctx, vni):
+    """Check if the vni is mapped to vrf
+    """
+
+    found = 0
+    db = ctx.obj['db']
+    vrf_table = db.get_table('VRF')
+    vrf_keys = vrf_table.keys()
+    if vrf_keys is not None:
+      for vrf_key in vrf_keys:
+        if ('vni' in vrf_table[vrf_key] and vrf_table[vrf_key]['vni'] == vni):
+           found = 1
+           break
+
+    if (found == 1):
+        print "VNI {} mapped to Vrf {}, Please remove VRF VNI mapping".format(vni, vrf_key)
```

Comment:
should it throw error or is it a warning? If it is error use ctx.fail().

Reply:
This is a utility function which is used by other functions and hence returning True or False.

Reply:
Seems that this line is the cause of errors during test phase. Probably because of Python version
See an error log after build, i.e. `make target/python-wheels/sonic_utilities-1.2-py3-none-any.whl`:

```
============================= test session starts ==============================
platform linux -- Python 3.7.3, pytest-3.10.1, py-1.7.0, pluggy-0.8.0
rootdir: /sonic/src/sonic-utilities, inifile: pytest.ini
plugins: cov-2.6.0
collected 142 items / 22 errors

==================================== ERRORS ====================================
________________ ERROR collecting tests/chassis_modules_test.py ________________
/usr/lib/python3/dist-packages/_pytest/python.py:450: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
/usr/lib/python3/dist-packages/py/_path/local.py:668: in pyimport
    __import__(modname)
<frozen importlib._bootstrap>:983: in _find_and_load
    ???
<frozen importlib._bootstrap>:967: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:668: in _load_unlocked
    ???
<frozen importlib._bootstrap>:638: in _load_backward_compatible
    ???
/usr/lib/python3/dist-packages/_pytest/assertion/rewrite.py:294: in load_module
    six.exec_(co, mod.__dict__)
tests/chassis_modules_test.py:9: in <module>
    import show.main as show
show/main.py:9: in <module>
    import utilities_common.cli as clicommon
E     File "/sonic/src/sonic-utilities/utilities_common/cli.py", line 359
E       print "VNI {} mapped to Vrf {}, Please remove VRF VNI mapping".format(vni, vrf_key)
E                                                                    ^
E   SyntaxError: invalid syntax
____________________ ERROR collecting tests/console_test.py ____________________
/usr/lib/python3/dist-packages/_pytest/python.py:450: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
/usr/lib/python3/dist-packages/py/_path/local.py:668: in pyimport
    __import__(modname)
<frozen importlib._bootstrap>:983: in _find_and_load
    ???
<frozen importlib._bootstrap>:967: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:668: in _load_unlocked
    ???
<frozen importlib._bootstrap>:638: in _load_backward_compatible
    ???
/usr/lib/python3/dist-packages/_pytest/assertion/rewrite.py:294: in load_module
    six.exec_(co, mod.__dict__)
tests/console_test.py:9: in <module>
    import config.main as config
config/main.py:23: in <module>
    import utilities_common.cli as clicommon
E     File "/sonic/src/sonic-utilities/utilities_common/cli.py", line 359
E       print "VNI {} mapped to Vrf {}, Please remove VRF VNI mapping".format(vni, vrf_key)
E                                                                    ^
E   SyntaxError: invalid syntax
```
... totally 22 times the same.

Reply:
Probably the reason is #1128

Reply:
Thanks for the comment. resolved now.